File: LMIS/LMIS.Web/Modules/BookManage/Books/BooksColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace LMIS.BookManage.Columns;

[ColumnsScript("BookManage.Books")]
[BasedOnRow(typeof(BooksRow), CheckNames = true)]
public class BooksColumns
{
    [EditLink, AlignRight]
    public long BookId { get; set; }
    [EditLink]
    public string BookName { get; set; }
    public string CN { get; set; }
    public string ISSN { get; set; }
    public string ISBN { get; set; }
    public string AuthorName { get; set; }
    public string PublisherName { get; set; }
    [Width(100)]
    [DateTimeFormatter(DisplayFormat = "yyyy/MM/dd")]
    public DateTime PublisheDate { get; set; }
    public int Page { get; set; }
    public decimal Price { get; set; }
    public long Inventory { get; set; }
    public short BookStatus { get; set; }
    public string BookImage { get; set; }
    [Width(150)]
    [DateTimeFormatter(DisplayFormat = "yyyy/MM/dd HH:mm:ss")]
    public DateTime CreateTime { get; set; }
    [Width(150)]
    [DateTimeFormatter(DisplayFormat = "yyyy/MM/dd HH:mm:ss")]
    public DateTime UpdateTime { get; set; }
}

File: LMIS/LMIS.Web/Modules/ServerTypes/BookManage/BooksColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { BooksRow } from "./BooksRow";

export interface BooksColumns {
    BookId: Column<BooksRow>;
    BookName: Column<BooksRow>;
    CN: Column<BooksRow>;
    ISSN: Column<BooksRow>;
    ISBN: Column<BooksRow>;
    AuthorName: Column<BooksRow>;
    PublisherName: Column<BooksRow>;
    PublisheDate: Column<BooksRow>;
    Page: Column<BooksRow>;
    Price: Column<BooksRow>;
    Inventory: Column<BooksRow>;
    BookStatus: Column<BooksRow>;
    BookImage: Column<BooksRow>;
    CreateTime: Column<BooksRow>;
    UpdateTime: Column<BooksRow>;
}

export class BooksColumns extends ColumnsBase<BooksRow> {
    static readonly columnsKey = 'BookManage.Books';
    static readonly Fields = fieldsProxy<BooksColumns>();
}